src/lib.rs

```diff
@@ -92,5 +92,25 @@ pub extern "C" fn __swift_bridge__null_pointer() -> *const std::ffi::c_void {
     std::ptr::null()
 }
 
+#[no_mangle]
+#[doc(hidden)]
+pub extern "C" fn __swift_bridge__equality_operator_for_RustStr(
+    lhs: string::RustStr,
+    rhs: string::RustStr,
+) -> bool {
+    if lhs.len != rhs.len {
+        return false;
+    }
+
+    unsafe {
```

Comment:
Another alternative that I thought of was:

```rust
unsafe { 
    std::slice::from_raw_parts(lhs.start, lhs.len) ==
    std::slice::from_raw_parts(rhs.start, rhs.len)
}
```

I was thinking that the stdlib might have an optimization that I was unfamiliar with, but I think that desugars to https://doc.rust-lang.org/1.52.1/src/core/slice/cmp.rs.html#69-75

Which looks like the same as what you're doing. So this looks good to me.